File: Data/CodeExamples/AutoFilterActions.cs
using System;
using System.Collections.Generic;
using System.Drawing;
using DevExpress.Spreadsheet;

namespace SpreadsheetExamples {
    public static class AutoFilterActions {

        static void ApplyFilter(IWorkbook workbook) {
            #region #ApplyFilter
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            #endregion #ApplyFilter
        }

        static void SortBySingleColumn(IWorkbook workbook) {
            #region #SortBySingleColumn
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Sort in descending order by the first column.
            worksheet.AutoFilter.SortState.Sort(0, true);

            #endregion #SortBySingleColumn
        }

        static void SortByMultipleColumns(IWorkbook workbook) {
            #region #SortByMultipleColumns
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Sort in descending order by the first and third columns.
            List<SortCondition> sortConditions = new List<SortCondition>();
            sortConditions.Add(new SortCondition(0, true));
            sortConditions.Add(new SortCondition(2, true));
            worksheet.AutoFilter.SortState.Sort(sortConditions);

            #endregion #SortByMultipleColumns
        }

        static void FilterByCondition(IWorkbook workbook) {
            #region #FilterByCondition
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data based on the condition SALES > 5000.
            worksheet.AutoFilter.Columns[2].ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan);

            #endregion #FilterByCondition
        }

        static void FilterByValue(IWorkbook workbook) {
            #region #FilterByValue
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data by a specific value.
            worksheet.AutoFilter.Columns[1].ApplyFilterCriteria("Mozzarella di Giovanni");

            #endregion #FilterByValue
        }

        static void FilterByValues(IWorkbook workbook) {
            #region #FilterByValues
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to worksheet range
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data by an array of values.
            worksheet.AutoFilter.Columns[1].ApplyFilterCriteria(new CellValue[] { "Mozzarella di Giovanni", "Gorgonzola Telino" });

            #endregion #FilterByValues
        }

        static void DynamicFilter(IWorkbook workbook) {
            #region #DynamicFilter
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data using dynamic criteria.
            worksheet.AutoFilter.Columns[2].ApplyDynamicFilter(DynamicFilterType.AboveAverage);

            #endregion #DynamicFilter
        }

        static void ReapplyFilter(IWorkbook workbook) {
            #region #ReapplyFilter
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data based on the condition SALES > 5000.
            worksheet.AutoFilter.Columns[2].ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan);

            // Change the data and reapply AutoFilter.
            worksheet["D3"].Value = 5000;
            worksheet.AutoFilter.ReApply();

            #endregion #ReapplyFilter
        }

        static void ClearFilter(IWorkbook workbook) {
            #region #ClearFilter
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Filter the data based on the condition SALES > 5000.
            worksheet.AutoFilter.Columns[2].ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan);

            // Clear the filter.
            worksheet.AutoFilter.Clear();

            #endregion #ClearFilter
        }

        static void DisableFilter(IWorkbook workbook) {
            #region #DisableFilter
            Worksheet worksheet = workbook.Worksheets["Regional sales"];
            workbook.Worksheets.ActiveWorksheet = worksheet;

            // Apply AutoFilter to a worksheet range.
            Range range = worksheet["B2:D9"];
            worksheet.AutoFilter.Apply(range);

            // Disable AutoFilter for the worksheet.
            worksheet.AutoFilter.Disable();

            #endregion #DisableFilter
        }
    }
}

File: Data/CodeExamples/AutoFilterActions.vb
Imports System
Imports System.Collections.Generic
Imports System.Drawing
Imports DevExpress.Spreadsheet

Namespace SpreadsheetExamples
	Public NotInheritable Class AutoFilterActions

		Private Sub New()
		End Sub
		Private Shared Sub ApplyFilter(ByVal workbook As IWorkbook)
'			#Region "#ApplyFilter"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

'			#End Region ' #ApplyFilter
		End Sub

		Private Shared Sub SortBySingleColumn(ByVal workbook As IWorkbook)
'			#Region "#SortBySingleColumn"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Sort in descending order by the first column.
			worksheet.AutoFilter.SortState.Sort(0, True)

'			#End Region ' #SortBySingleColumn
		End Sub

		Private Shared Sub SortByMultipleColumns(ByVal workbook As IWorkbook)
'			#Region "#SortByMultipleColumns"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Sort in descending order by the first and third columns.
			Dim sortConditions As New List(Of SortCondition)()
			sortConditions.Add(New SortCondition(0, True))
			sortConditions.Add(New SortCondition(2, True))
			worksheet.AutoFilter.SortState.Sort(sortConditions)

'			#End Region ' #SortByMultipleColumns
		End Sub

		Private Shared Sub FilterByCondition(ByVal workbook As IWorkbook)
'			#Region "#FilterByCondition"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data based on the condition SALES > 5000.
			worksheet.AutoFilter.Columns(2).ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan)

'			#End Region ' #FilterByCondition
		End Sub

		Private Shared Sub FilterByValue(ByVal workbook As IWorkbook)
'			#Region "#FilterByValue"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data by a specific value.
			worksheet.AutoFilter.Columns(1).ApplyFilterCriteria("Mozzarella di Giovanni")

'			#End Region ' #FilterByValue
		End Sub

		Private Shared Sub FilterByValues(ByVal workbook As IWorkbook)
'			#Region "#FilterByValues"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to worksheet range
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data by an array of values.
			worksheet.AutoFilter.Columns(1).ApplyFilterCriteria(New CellValue() { "Mozzarella di Giovanni", "Gorgonzola Telino" })

'			#End Region ' #FilterByValues
		End Sub

		Private Shared Sub DynamicFilter(ByVal workbook As IWorkbook)
'			#Region "#DynamicFilter"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data using dynamic criteria.
			worksheet.AutoFilter.Columns(2).ApplyDynamicFilter(DynamicFilterType.AboveAverage)

'			#End Region ' #DynamicFilter
		End Sub

		Private Shared Sub ReapplyFilter(ByVal workbook As IWorkbook)
'			#Region "#ReapplyFilter"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data based on the condition SALES > 5000.
			worksheet.AutoFilter.Columns(2).ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan)

			' Change the data and reapply AutoFilter.
			worksheet("D3").Value = 5000
			worksheet.AutoFilter.ReApply()

'			#End Region ' #ReapplyFilter
		End Sub

		Private Shared Sub ClearFilter(ByVal workbook As IWorkbook)
'			#Region "#ClearFilter"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Filter the data based on the condition SALES > 5000.
			worksheet.AutoFilter.Columns(2).ApplyCustomFilter(5000, FilterComparisonOperator.GreaterThan)

			' Clear the filter.
			worksheet.AutoFilter.Clear()

'			#End Region ' #ClearFilter
		End Sub

		Private Shared Sub DisableFilter(ByVal workbook As IWorkbook)
'			#Region "#DisableFilter"
			Dim worksheet As Worksheet = workbook.Worksheets("Regional sales")
			workbook.Worksheets.ActiveWorksheet = worksheet

			' Apply AutoFilter to a worksheet range.
			Dim range As Range = worksheet("B2:D9")
			worksheet.AutoFilter.Apply(range)

			' Disable AutoFilter for the worksheet.
			worksheet.AutoFilter.Disable()

'			#End Region ' #DisableFilter
		End Sub
	End Class
End Namespace